app/api/ada/adaLocalStorage.js

```diff
@@ -56,6 +83,19 @@ export function getLastBlockNumber(): number {
   return Number(lastBlockNum);
 }
 
+/* Last block number storage */
+
+export function saveLastReceiveAddressIndex(index: number): void {
+  const stored: LocalStorageWallet = _getFromStorage(storageKeys.WALLET_KEY);
+  stored.lastReceiveAddressIndex = index;
+  updateWallet(stored);
+}
```

Comment:
This looks a bit weird that usage of `storageKeys.WALLET_KEY` is spread across multiple functions, but then there's `updateWallet` function that incapsulates that key. It appears as if `updateWallet` does something very different from `_getFromStorage`, even tho this is basically an inverse function, but with one argument hidden.

My advice would be to define `getWalletFromStorage` and optionally rename `updateWallet` into `updateWalletInStorage`. Or remove `updateWallet` and always use `storageKeys.WALLET_KEY` explicitly 🤔